server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -402,130 +413,601 @@ describe("Routerlicious", () => {
+							clients.forEach((client) => {
+								client.signalPromise = createSignalPromise(client);
+							});
+
+							clients[1].socket.send("submitSignal", clients[1].clientId, [
+								"second signal",
+							]);
+
+							userSignals = await Promise.all(
+								clients.map((client) => client.signalPromise),
+							);
 
-						const promises = [
-							new Promise<void>((resolve) => {
-								firstSocket.on("signal", (signal: ISignalMessage) => {
+							clients.forEach((client, index) => {
+								assert.equal(
+									client.signalCount,
+									2,
+									`User ${index + 1} should have received 2 signals`,
+								);
+								assert.equal(
+									userSignals[index].content,
+									"second signal",
+									`User ${index + 1} signal content mismatch`,
+								);
+							});
+						});
+
+						it("can handle invalid clientID", async () => {
+							clients[0].socket.send("submitSignal", "invalidClientID", [
+								stringSignalContent,
+							]);
+							await Promise.all(clients.map((client) => client.signalPromise));
+							clients.forEach((client, index) => {
+								assert.equal(
+									client.signalCount,
+									0,
+									`User ${index + 1} should not have received any signals`,
+								);
+							});
+						});
+
+						it("can handle a receiving client disconnect", async () => {
+							clients[1].socket.disconnect();
+
+							// Ignore the signal from the disconnected client
+							clients.forEach((client) => {
+								client.signalPromise = createSignalPromise(client);
+							});
+
+							clients[0].socket.send("submitSignal", clients[0].clientId, [
+								stringSignalContent,
+							]);
+
+							await Promise.all(
+								clients.map((client) => {
+									client.signalPromise;
+								}),
+							);
+
+							clients.forEach((client, index) => {
+								if (index === 1) {
 									assert.equal(
-										signalContent,
-										signal.content,
-										"Signal content mismatch",
+										client.signalCount,
+										0,
+										`User ${index + 1} should not have received any signals`,
 									);
+								} else {
 									assert.equal(
-										firstConnectMessage.clientId,
-										signal.clientId,
-										"Client ID mismatch",
+										client.signalCount,
+										1,
+										`User ${index + 1} should have received 1 signal`,
+									);
+								}
+							});
+						});
+
+						it("can handle a sending client disconnect", async () => {
+							clients[0].socket.send("submitSignal", clients[0].clientId, [
+								stringSignalContent,
+							]);
+							clients[0].socket.disconnect();
+
+							await Promise.all(
+								clients.map((client) => {
+									client.signalPromise;
+								}),
+							);
+
+							clients.forEach((client, index) => {
+								assert.equal(
+									client.signalCount,
+									1,
+									`User ${index + 1} should have received 1 signal`,
+								);
+							});
+						});
```

Comment:
What is this trying to test? Can you make this fail while other tests are passing?
My suspicion is that the timing is such that there isn't anything happening here of consequence.